Makefile.core.mk

```diff
@@ -264,10 +256,23 @@ gen-code: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and
 
 .PHONY: gen-charts
 gen-charts: ## Pull charts from maistra/istio repository
-	hack/download-charts.sh v${MINOR_VERSION} https://github.com/${ISTIO_REPOSITORY} ${ISTIO_COMMIT_30}
+	@yq eval 'to_entries | .[] | "./hack/download-charts.sh " + .key + " " + .value.repo + " " + .value.commit + " " + ((.value.charts // []) | join(" "))' < versions.yaml | sh
+	@hack/copy-crds.sh "resources/$$(yq eval 'keys | .[0]' versions.yaml)/charts"
+
+.PHONY: update-version-list
+update-version-list: ## Updates the urn:alm:descriptor:com.tectonic.ui:select entries in istio_types.go to match the supported versions of the Helm charts
+	@echo "Updating version list in istio_types.go..."
+	@selectValues=$$(yq e 'keys | .[] | ", \"urn:alm:descriptor:com.tectonic.ui:select:" + . + "\""' versions.yaml | tr -d '\n'); \
+		versionsEnum=$$(yq e 'keys | .[]' versions.yaml | tr '\n' ';' | sed 's/;$$//g'); \
+		versions=$$(yq e 'keys | .[]' versions.yaml | tr '\n' ',' | sed -e 's/,/, /g' -e 's/, $$//g'); \
+		sed -i -E \
+			-e "s/(\/\/ \+operator-sdk:csv:customresourcedefinitions:type=spec,order=1,displayName=\"Istio Version\",xDescriptors=\{.*fieldGroup:General\")[^}]*(})/\1$$selectValues}/g" \
+			-e "s/(\/\/ \+kubebuilder:validation:Enum=)(.*)/\1$$versionsEnum/g" \
+			-e "s/(\/\/ \Must be one of:)(.*)/\1 $$versions./g" \
+			api/v1alpha1/istio_types.go
 
 .PHONY: gen ## Generate everything
-gen: controller-gen gen-manifests gen-code gen-charts bundle
+gen: update-version-list controller-gen gen-manifests gen-code gen-charts bundle
```

Comment:
If we are keeping this target (i.e., not merging it with `gen-charts`, I think it should be moved to after `gen-charts`.

Reply:
see my previous comment; update-version-list must be invoked before controller-gen.